src/streaming/controllers/BufferController.js

```diff
@@ -347,17 +313,16 @@ MediaPlayer.dependencies.BufferController = function () {
 
         /* prune buffer on our own in background to avoid browsers pruning buffer silently */
         pruneBuffer = function() {
-            var bufferToPrune = 0,
+            var start = buffer.buffered.length ? buffer.buffered.start(0) : 0,
                 currentTime = this.playbackController.getTime(),
-                currentRange = this.sourceBufferExt.getBufferRange(buffer, currentTime);
-
-            // we want to get rid off buffer that is more than x seconds behind current time
-            if (currentRange !== null) {
-                bufferToPrune = currentTime - currentRange.start - MediaPlayer.dependencies.BufferController.BUFFER_TO_KEEP;
-                if (bufferToPrune > 0) {
-                    isPruningInProgress = true;
-                    this.sourceBufferExt.remove(buffer, 0, Math.round(currentRange.start + bufferToPrune), mediaSource);
-                }
+                bufferToPrune = currentTime - start - MediaPlayer.dependencies.BufferController.BUFFER_TO_KEEP;
+
+            // we want to get rid off buffer that is more than x
+            // seconds behind current time, but no pruning once it's
+            // finished.
+            if (!isPruningInProgress && mediaSource.readyState !== "ended" && bufferToPrune > MINIMUM_BUFFER_TO_PRUNE) {
```

Comment:
I don't think we should introduce here another variable MINIMUM_BUFFER_TO_PRUNE. This overwrites the behavior of BUFFER_TO_KEEP when it is set lower than 20s. 
It's better to fix the pruning interval -> see PR #898